### ISSUE TITLE
Enable running unittests safely in parallel by giving them independent workspaces

### ISSUE THREAD

**Issue by katrinabrock** [stan-dev/cmdstanr#1025]
(likely low priority)

**Is your feature request related to a problem? Please describe.**
Currently unit tests take ~1hr to run. Running in parallel is risky because multiple scripts recompile the same binary with different params and thus could potentially conflict.

Additionally, this risks previous tests polluting future tests. 

**Describe the solution you'd like**
First idea that comes to mind is giving each script a separate directory where its stan files and binaries live.

**Comment by WardBrian** [comment]
Just as a warning, the cmdstan folder itself can also have issues if you are trying to compile the same file (usually main.o and its variants) multiple times in parallel.
